evidence/pool.go

```diff
@@ -520,10 +532,19 @@ func (evpool *Pool) removeEvidenceFromList(
 	}
 }
 
-func (evpool *Pool) updateState(state sm.State) {
-	evpool.mtx.Lock()
-	defer evpool.mtx.Unlock()
-	evpool.state = state
+// flushConsensusBuffer moves the evidence produced from consensus into the evidence pool
+// and list so that it can be broadcasted and proposed
+func (evpool *Pool) flushConsensusBuffer() {
```

Comment:
nit: I would've added a write-lock here instead of using it in `Pool#Update`. That way, it becomes:

```go
evpool.flushConsensusBuffer()
evpool.updateState(state)
```

I say this because we have `updateState` already that does write-locking, but now `Pool#Update` does it :-/

Reply:
Yeah I guess this makes sense to avoid someone in the future from using `evpool.flushConsensusBuffer()` without taking out a lock

Reply:
I also wasn't too sure if the best practice was to create another lock or use the existing lock on `state` for `consensusBuffer`

Reply:
since we have auxiliary logic (method) that does all the business logic, I believe we should place the lock here. This isn't necessarily critical though 👍